src/towncrier/_writer.py

```diff
@@ -8,7 +8,22 @@
 
 from __future__ import annotations
 
+import sys
+
 from pathlib import Path
+from typing import Any
+
+
+if sys.version_info < (3, 10):
```

Comment:
I think that this needs a comment.
Once we no longer need to support 3.9 or older, this code can be removed.

Reply:
Fair, though in my mind, the code is pretty explicit about which versions rely on the compatibility code. I was tempted to add a comment just to explain why, but my feeling was that the why could be readily derived by removing the compatibility and testing what fails. Still, it's faster to read a comment.

Added in 8b821dc.